nemo-physical/src/function/definitions/numeric.rs

```diff
@@ -499,6 +500,81 @@ impl NaryFunction for BitXor {
     }
 }
 
+/// Bitwise Left shift
+///
+/// Returns `None` if the input parameter pair are not integers or no input parameters are given.
+#[derive(Debug, Copy, Clone)]
+pub struct BitShiftLeft;
+impl BinaryFunction for BitShiftLeft {
+    fn evaluate(
+        &self,
+        parameter_first: AnyDataValue,
+        parameter_second: AnyDataValue,
+    ) -> Option<AnyDataValue> {
+        if let Some(pair) = NumericPair::from_any_pair_cast(&parameter_first, &parameter_second) {
+            match pair {
+                NumericPair::Integer(value, base) => numeric_bitwise_shl(value, base),
+                _ => None,
+            }
+        } else {
+            None
+        }
```

Comment:
Even if you don't want to get rid of the explict `let` here (for whatever reason), the `if let Some(…) {…} else { None } ` is just noise that could be
```rust
let pair = NumericPair::from_any_pair_cast(&parameter_first, &parameter_second)?;
match pair { … }
```
which doesn't require looking at different branches.

I'd still prefer
```rust
        match NumericPair::from_any_pair_cast(&parameter_first, &parameter_second)? {
            NumericPair::Integer(value, base) => numeric_bitwise_shl(value, base),
            _ => None,
        }
```